x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -339,4 +340,27 @@ public String toString() {
                 completed.size(),
                 stageToKeys);
     }
+
+    private static long timestampDiffNanos(Object ts1, Object ts2) {
```

Comment:
Maybe it would be good to have some dedicated unit tests for this method. Does it support diffing over the fully supported range of timestamps? Also, I can imagine the method is vulnerable to long overflow issues.

Reply:
This has been now refactored and mostly covered by OrdinalTests.java.